File: python/order.py
class Order:

    def __init__(self, order_type, ticker, price, quantity):
        self.order_type = order_type.upper()    # "BUY" or "SELL"
        self.ticker = ticker                    # Ticker symbol index (0 to 1023)
        self.quantity = quantity                # Number of shares
        self.price = price                      # Price per share
        self.next = None                        # Reference for linked list node  

    def __repr__(self):
        return f"[Ticker {self.ticker}] {self.order_type} {self.quantity} @ ${self.price}"

File: java/StockEngine/src/Order.java
import java.util.concurrent.atomic.AtomicInteger;
import java.util.concurrent.atomic.AtomicReference;

public class Order {
    public enum OrderType { BUY, SELL }

    public final OrderType orderType;
    public final int ticker;
    public final double price;
    public final AtomicInteger quantity;
    public final AtomicReference<Order> next;

    public Order(OrderType orderType, int ticker, double price, int quantity) {
        this.orderType = orderType;
        this.ticker = ticker;
        this.price = price;
        this.quantity = new AtomicInteger(quantity);
        this.next = new AtomicReference<>(null);
    }

    @Override
    public String toString() {
        return "[Ticker " + ticker + "] " + orderType + " " + quantity.get() + " @ $" + price;
    }
}
